### DISCUSSION TITLE
Extract checkboxes from pdf

### DISCUSSION THREAD

**Discussion by Atul997** [jsvine/pdfplumber#474] [Ask for help with specific PDFs]
How can I extract marked check-boxes and corresponding text from the pdf file. I tried `extract_words()` working fine. For check-boxes it returns only the ticked as `X` and corresponding text.  What should I do to extract only check-boxes ?

[Residential Land Appraisal Form.PDF](https://github.com/jsvine/pdfplumber/files/6835800/Residential.Land.Appraisal.Form.PDF)

This is the sample file from which I have to extract check boxes

**Comment by jsvine** [comment]
Hi @Atul997, and interesting challenge. For this particular PDF, each checkbox appears to be composed of four lines. E.g., the following code:

```python
colors = [ "red", "green", "orange", "blue" ]
page = pdf.pages[3]
im = page.to_image(resolution=300)
for i, line in enumerate(page.lines):
	im.draw_line(line, stroke_width=4, stroke=colors[i % 4])
```

... produces this output (which I have cropped):

<img width="1225" alt="Screen Shot 2021-08-11 at 11 04 38 PM" src="https://user-images.githubusercontent.com/534702/129132305-fc5c12ec-9a98-489c-a168-07248421775f.png">

So, it seems likely that you use the data in `page.lines`  to identify the positions of all the checkboxes.

**Reply by abhijithnair1** [reply]
@jsvine Is it possible to tell which boxes have tick marks and which doesn't (i.e, which are empty and which are not empty). Also is it possible to associate that checkpoint with relevant form key?

**Reply by jsvine** [reply]
In cases like these — i.e., where the checkbox is drawn from graphical components, rather than as part of an [AcroForm](https://github.com/jsvine/pdfplumber?tab=readme-ov-file#extracting-form-values) — this would require custom coding based on the particular PDF. Using the PDF above as an example, I'd analyze the positions of those checkboxes, and then examine each to see whether there are tick marks (which are also encoded as `line` objects) inside them.
